core/src/test/java/org/apache/iceberg/TestWapWorkflow.java

```diff
@@ -684,4 +685,87 @@ public void testNonWapCherrypick() {
           table.manageSnapshots().cherrypick(firstSnapshotId).commit();
         });
   }
+
+  @Test
+  public void testWithExpiringWapThenCherrypick() {
+    table.newAppend()
+            .appendFile(FILE_A)
+            .commit();
+    // first WAP commit
+    table.newAppend()
+            .appendFile(FILE_B)
+            .set(SnapshotSummary.STAGED_WAP_ID_PROP, "123456789")
```

Comment:
The tests shouldn't use WAP properties at all. Correctness depends on where commits were picked and should ignore the WAP ID. The reason is that two commits with the same WAP id don't contain the same files. They are logically the same data, not physically the same files. The only equivalence that matters is when a snapshot itself has been picked because the files are physically the same.

Reply:
@mehtaashish23 - Can you add a UT for each scenario we are trying to handle? It might help debug in case we see rogue behavior. Thanks.